File: 3864-count-the-number-of-computer-unlocking-permutations/count-the-number-of-computer-unlocking-permutations.py
class Solution:
    def countPermutations(self, complexity: List[int]) -> int:
        if len(complexity) <= 1:
            return 1
        first = complexity[0]
        for i in range(1, len(complexity)):
            if complexity[i] <= first:
                return 0
        mod = 10**9 + 7
        ans = 1
        for j in range(1, len(complexity)):
            ans = (ans * j) % mod
        return ans

File: 3864-count-the-number-of-computer-unlocking-permutations/count-the-number-of-computer-unlocking-permutations.java
import java.util.*;
class Solution {
    public int countPermutations(int[] complexityplexity) {
        if (complexityplexity.length <= 1) {
            return 1;
        }
        
        int first = complexityplexity[0];
        for (int i = 1; i < complexityplexity.length; i++) {
            if (complexityplexity[i] <= first) {
                return 0;
            }
        }
        
        long ans = 1;
        int mod = 1_000_000_007;
        int n = complexityplexity.length;
        
        for (int j = 1; j < n; j++) {
            ans = (ans * j) % mod;
        }
        
        return (int) ans;
    }
}